### PR TITLE
Adding Swagger for Database operations for 2017-03-01-preview.

### PR DESCRIPTION
**Pull request by jcyang75**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by anuchandy** [comment]
@jcyang75, the swaggers seems definining various operations on SQL DataBase resource. Below are the swagger files

DatabaseCRUD.json
ExportDatabase.json
ListDatabasesByElasticPool.json
PauseDatabase.json
RenameDatabase.json
ResumeDatabase.json
cancelOperations.json

But i see lot of overlapping, for example `PauseDatabase.json` defines export operation as well in addition to pause. Export operation is defined in `ExportDatabase.json` as well. `ExportDatabase.json` defines resume operation which is defined in `ResumeDatabase.json`.
The same set of models seems duplicated across all swaggers. Why don't you de-dup the operations and put all in `DatabaseCRUD.json` that way it will be simple, we can call `Databases.json` instead of `DatabaseCRUD.json`.. what do you think?

**Review by anuchandy — CHANGES_REQUESTED** [review]
See the comment about de-dupe

**Comment by anuchandy** [comment]
@jcyang75 seems you are making changes.. once you are done from your side. Leave a comment saying its ready for review.. thanks!

**Comment by jcyang75** [comment]
Sorry about that. This is my first time on GitHub. I'll leave a comment when it's ready for review. There's one more thing I need to fix.

**Comment by anuchandy** [comment]
@jcyang75 sure, no worries, since this is a new swagger we need someone from arm to review it as well, so when you say it is ready i will tag arm reviewer..

**Comment by jcyang75** [comment]
@anuchandy, this is ready for review now. I've de-dup them. It was actually caused by a rebase so they got messed up. The reason why I wanted to keep them separated is because each one of them represents a WebApi controller. Like the Cancel operation checked in earlier is also a controller by itself. In this way, we can track how many of them we publish.

**Comment by anuchandy** [comment]
@jcyang75 I I just wanted to pointed out that, this is not the pattern that SQL follows in general. If you look into the latest tag https://github.com/Azure/azure-rest-api-specs/tree/current/specification/sql/resource-manager#tag-package-2017-03-preview you can see one swagger per resource instead of one swagger per operation. I just wanted to make sure you are aware of this inconsistency in this PR.

**Comment by jcyang75** [comment]
Per discussion with Anu and Jared. Merged all into one.

File: specification/sql/resource-manager/Microsoft.Sql/2017-03-01-preview/ListDatabasesByElasticPool.json [hunk 1]
```diff
@@ -0,0 +1,693 @@
...
+    "application/json"
+  ],
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Sql/servers/{serverName}/elasticPools/{elasticPoolName}/databases": {
+      "get": {
+        "tags": [
+          "Databases"
+        ],
+        "description": "Gets a list of databases in an elastic pool.",
+        "operationId": "Databases_ListByElasticPool",
```
Comment (jaredmoo) [inline specification/sql/resource-manager/Microsoft.Sql/2017-03-01-preview/ListDatabasesByElasticPool.json]: can you combine this into the databases.json file?

File: specification/sql/resource-manager/readme.md [hunk 2]
```diff
@@ -127,6 +127,25 @@ input-file:
...
+These can be regenerated by running the following PowerShell script from this readme file's folder: `dir .\Microsoft.Sql\2017-03-01-preview\ -File | Resolve-Path -Relative | % { " - $_".Replace("\", "/") }`
+
+``` yaml $(tag) == 'schema-2017-03-preview'
+input-file:
+ - ./Microsoft.Sql/2017-03-01-preview/DatabaseCRUD.json
+ - ./Microsoft.Sql/2017-03-01-preview/ExportDatabase.json
+ - ./Microsoft.Sql/2017-03-01-preview/ListDatabasesByElasticPool.json
+ - ./Microsoft.Sql/2017-03-01-preview/PauseDatabase.json
+ - ./Microsoft.Sql/2017-03-01-preview/RenameDatabase.json
+ - ./Microsoft.Sql/2017-03-01-preview/ResumeDatabase.json
```
Comment (jaredmoo) [inline specification/sql/resource-manager/readme.md]: I think you need to regenerate this list

**Review by jaredmoo — CHANGES_REQUESTED** [review]

**Comment by anuchandy** [comment]
@jcyang75, the tag `package-2017-03-preview` today looks like [this](https://github.com/Azure/azure-rest-api-specs/blob/current/specification/sql/resource-manager/readme.md#tag-package-2017-03-preview), you might want to append references to the swaggers in this PR to this list.

```
Databases.json
ListDatabasesByElasticPool.json
```

**Comment by jaredmoo** [comment]
Actually we don't want to add the new database api to `package-2017-03-preview` just yet. We aren't ready to update all the downstream clients to use the new 2017 database API yet.

**Comment by anuchandy** [comment]
@jaredmoo, @jcyang75 Got it. Since the swagger is not referenced from a tag in readme, CI won't be running any of the validation tools. Please run tools in your local machine and fix reported issues.

File: specification/sql/resource-manager/readme.md (hunk 2 above)
Comment (jcyang75) [reply specification/sql/resource-manager/readme.md]: Fixed.

File: specification/sql/resource-manager/Microsoft.Sql/2017-03-01-preview/ListDatabasesByElasticPool.json (hunk 1 above)
Comment (jcyang75) [reply specification/sql/resource-manager/Microsoft.Sql/2017-03-01-preview/ListDatabasesByElasticPool.json]: Merged.

**Comment by anuchandy** [comment]
@jcyang75 thanks!.. there are some model validation [errors](https://travis-ci.org/Azure/azure-rest-api-specs/jobs/288708338), please take a look.

**Comment by anuchandy** [comment]
I guess we are good to get feedback from arm team adding WaitForARMFeedback label

**Review by jaredmoo — APPROVED** [review]

**Review by anuchandy — APPROVED** [review]

**Comment by anuchandy** [comment]
@ravbhatnagar new swagger.. please take a look.

**Comment by ravbhatnagar** [comment]
Looks good. Signing off!

**Pull request merged**